File: memory.java
class Wrapper {
    int value;

    Wrapper(int value) {
        this.value = value;
    }
}

public class MemoryDemo {
    public static void main(String[] args) {
        Wrapper obj = new Wrapper(100); // Heap allocation
        printValue(obj); // Passing reference
        System.out.println("Still accessible " + obj.value);
    }

    static void printValue(Wrapper obj) {
        System.out.println("Value: " + obj.value);
    }
}

File: memory.cpp
#include <iostream>

class Wrapper {
public:
    int value;
    Wrapper(int val) : value(val) {}
}

void printValue(const Wrapper* obj) { // Pass by pointer (read-only)
    std::cout << "Value: " << obj->value << std::endl;
}

int main() {
    Wrapper *obj = new Wrapper(100); // Heap allocation
    printValue(obj); // Passp ointer
    std::cout << "Still accessible" << obj->value << std::endl;
    delete obj; // Manual deallocation
}